### DISCUSSION TITLE
Orhard Core and Keyclock

### DISCUSSION THREAD

**Discussion by Morgfrimen** [OrchardCMS/OrchardCore#11918] [Q&A]
Good afternoon, colleagues! I managed to connect to Orchard Core Keyclock (OpenID). But in the course of work, a question arose - how do I log out of my account? It's just not in the admin panel... Or what do I not understand?

**Comment by Morgfrimen** [comment]
Signed Out does not work as usual in OrchardCore.OpenId.Client ...

**Comment by Skrypt** [comment]
Can you show a screencast with a tool like ScreenToGif so that we can see what you mean exactly?

**Reply by Morgfrimen** [reply]
Yes, sorry for not being accurate.
While I was solving the problem, I found the problem, but I still don't know how to solve it. Currently, there is authorization via KeyClock (identification server). User roles and roles in Orchard Core are configured in it. Everything works fine except exiting the app. It turns out that Orchard Core logs out of its account, but the Keyclock cookies remain in place and the next time I try to log in through Keyclock, orchard core immediately authorizes me.
Hoping to fix the situation, I redefined the logoff method of the AccountController controller in Orchard Core.Users.

```csharp
        [HttpPost]
        [ValidateAntiForgeryToken]
        public async Task<IActionResult> LogOff()
        {
            await _signInManager.SignOutAsync();

            var prop = await _signInManager.Context.AuthenticateAsync();
            prop.Properties!.RedirectUri = "{domen}/auth/realms/{proj}/protocol/openid-connect/logout";

            try
            {
                var authenticationSchemes = await _signInManager.GetExternalAuthenticationSchemesAsync();
                
                foreach (var authenticationScheme in authenticationSchemes)
                {
                    await HttpContext.SignOutAsync(authenticationScheme.Name,prop.Properties);
                }

                await HttpContext.SignOutAsync(IdentityConstants.ExternalScheme);
            }
            catch
            {
                _logger.LogInformation(4, "user error logout");
            }
            _logger.LogInformation(4, "user logout");
            return Redirect("~/");
        }
```
The method works, I get into it during execution. But for some reason he still doesn't want to delete the KeyClock cookies. And I noticed that
```csharp
await _signInManager.GetExternalAuthenticationSchemesAsync()
```
return "OpenIdConnection", but "Identity.Application" default in _signInManager.

I don't understand how I get out of the KeyCloke into the Orchard Core. What didn't I do?
